### PR TITLE
RavenDB-13181 Counters replication discrepancies

### PR DESCRIPTION
**Pull request by karmeli87**
The underlying issue is that we resotred not in the insertion order in which the document, revisions, coutners, etc. were putted in the original database.
In the restored database this could cause that a document with higher change vector will have a lower local etag a therefore will skip the replication of doucments with lower change vector.

- Remove keep original change vector
- Insert the counters by insertion order

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/CountersHandler.cs [hunk 1]
```diff
@@ -39,12 +39,13 @@ public class ExecuteCounterBatchCommand : TransactionOperationsMerger.MergedTran
             private readonly bool _replyWithAllNodesValues;
             private readonly bool _fromEtl;
             private readonly bool _fromSmuggler;
-            private readonly Dictionary<string, List<CounterOperation>> _dictionary;
+            private readonly List<CounterOperation> _list;
+            //private readonly Dictionary<string, List<CounterOperation>> _dictionary;
```
Comment (grisha-kotler) [inline src/Raven.Server/Documents/Handlers/CountersHandler.cs]: can be removed

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs [hunk 2]
```diff
@@ -88,6 +88,7 @@ public class CounterOperation
         public long Delta;
 
         internal string ChangeVector;
+        public string DocumentId;
```
Comment (ayende) [inline src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs]: Shouldn't this be internal?

Comment (karmeli87) [reply src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs]: It fails on the TxReplay, since serializing this ignore the internal.

Comment (ayende) [reply src/Raven.Client/Documents/Operations/Counters/CounterBatch.cs]: Can we fix the serialization, instead?
This is sitting on the client, and I don't want it exposed to users

**Review by aviv86 — APPROVED** [review]

**Review by ppekrol — APPROVED** [review]

**Review by ayende — APPROVED** [review]

**Pull request merged**